demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -115,6 +149,83 @@
  */
 #define DELAY_BETWEEN_DEMO_RETRY_ITERATIONS_S    ( 5 )
 
+/**
+ * @brief Length in bytes of hash digest.
+ */
+#define SHA256_HASH_DIGEST_LENGTH                ( ( ( uint16_t ) 64 ) )
+
+/**
+ * @brief Buffer Length for AWS IOT Credentials.
+ */
+#define CREDENTIAL_BUFFER_LENGTH                 2048U
+
+/**
+ * @brief AWS Service name to send HTTP request using SigV4 library.
+ */
+#define AWS_SERVICE_NAME                         "s3"
+
+/**
+ * @brief Full AWS S3 server endpoint.
+ */
+#define AWS_S3_OBJECT_URL                       \
+    "https://"                                  \
+    AWS_S3_BUCKET_NAME "." AWS_SERVICE_NAME "." \
+    AWS_S3_BUCKET_REGION  ".amazonaws.com/"     \
+    AWS_S3_OBJECT_NAME
+
+/**
+ * @brief Full credentials endpoint including role alias.
+ */
+#define AWS_IOT_CREDENTIAL_PROVIDER_FULL_ENDPOINT         \
```

Comment:
This is not an endpoint but a full-fledged URL
```suggestion
#define AWS_IOT_CREDENTIAL_PROVIDER_URL \
```